### PR TITLE
doc: add release checklist

### PR DESCRIPTION
**Pull request by ajeddeloh**

### CODE REVIEW

File: doc/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ ```
+ * [ ] Sign the release artifacts by running
+ ```
+ gpg -u gpg --local-user 0xCDDE268EBB729EC7 --detach-sign --armor <path to artifact>
+ ```
+   for each release artifact. Do _not_ try to sign all of them at once by globbing. If you do, gpg will
+   sign the combination of all the release artifacts instead of each one individually.
+ * [ ] Create a draft release on Github for the tag and upload all the release artifacts and
+       their signatures. Copy and paste the release notes from NEWS here as well.
+ * [ ] Get someone to review the draft release
```
Comment (dgonyeo) [inline doc/release_checklist.md]: There should be an extra step here of "tag the resulting docker image at `quay.io/coreos/ct` with the new version.", so something like:
```
docker pull quay.io/coreos/ct:v0.6.1-11-g90d21c6
docker tag quay.io/coreos/ct:v0.6.1-11-g90d21c6 quay.io/coreos/ct:v0.7.0
docker push quay.io/coreos/ct:v0.7.0
```

CI for the release PR runs before the git tag exists, so it won't be tagged with the new version.

Comment (dgonyeo) [reply doc/release_checklist.md]: I just did this for v0.7.0, so the instructions should read:

> Once the new version of ct has been pushed to the docker hub, run (versions and hashes will be different):
> ```
> docker pull quay.io/coreos/ct:v0.7.0-2-gac076a8
> docker tag quay.io/coreos/ct:v0.7.0-2-gac076a8 quay.io/coreos/ct:v0.7.0
> docker push quay.io/coreos/ct:v0.7.0
> ```

**Comment by ajeddeloh** [comment]
Updated to include docker instructions

File: doc/release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * [ ] Publish!
+ * [ ] Tag the docker image that was automatically built when the NEWS PR was merged
+   * [ ] Run `git describe` to determine which docker container to tag.
+   * [ ] Run `docker pull quay.io/coreos/ct:<git-descibe-output>`
+   * [ ] Run `docker tag quay.io/coreos/ct:<git-describe-output> quay.io/coreos/ct:vX.Y.Z`
+   * [ ] Run `docker push quay.io/coreos/ct:v0.7.0`
```
Comment (dgonyeo) [inline doc/release_checklist.md]: nit: `vX.Y.Z`

**Pull request merged**